module/x/gravity/keeper/attestation.go

```diff
@@ -331,6 +331,10 @@ func (k Keeper) GetLastObservedEthereumBlockHeight(ctx sdk.Context) types.LastOb
 // SetLastObservedEthereumBlockHeight sets the block height in the store.
 func (k Keeper) SetLastObservedEthereumBlockHeight(ctx sdk.Context, ethereumHeight uint64) {
 	store := ctx.KVStore(k.storeKey)
+	previous := k.GetLastObservedEthereumBlockHeight(ctx)
+	if previous.EthereumBlockHeight > ethereumHeight {
```

Comment:
Why are we not considering `prev.EthBlockHeight == ethHeight` here? Wouldn't it be an issue to set the same height twice (e.g. replay attack)?

Reply:
well multiple events can happen in the same block currently I think we just set it twice, this is not hte case for the event nonce though.